### PR TITLE
make the package v3router/judger test coverage rate reach 80%

### PR DESCRIPTION
**Pull request by Mulavar**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by LaurenceLiZhixin — APPROVED** [review]

File: cluster/router/v3router/judger/attachment_match_judger.go [hunk 1]
```diff
@@ -29,45 +29,34 @@ type AttachmentMatchJudger struct {
...
-				continue
-			}
-			// exist this key
-			str, ok := invAttaValue.(string)
-			if !ok {
-				return false
-			}
-			strJudger := NewStringMatchJudger(v)
-			if !strJudger.Judge(str) {
+	if amj.DubboContext != nil && !judge(amj.DubboContext, invAttaMap) {
```
Comment (gaoxinge) [inline cluster/router/v3router/judger/attachment_match_judger.go:36]: directly use `return amj.DubboContext == nil || judge(amj.DubboContext, invAttaMap)`

File: cluster/router/v3router/judger/attachment_match_judger.go [hunk 2]
```diff
@@ -29,45 +29,34 @@ type AttachmentMatchJudger struct {
...
 			}
+			continue
+		}
+		// exist this key
+		str, ok := invAttaValue.(string)
+		if !ok {
+			return false
+		}
+		strJudger := NewStringMatchJudger(v)
+		if !strJudger.Judge(str) {
```
Comment (gaoxinge) [inline cluster/router/v3router/judger/attachment_match_judger.go:58]: directly use `return strJudger.Judge(str)`

Comment (gaoxinge) [inline cluster/router/v3router/judger/attachment_match_judger.go:58]: directly use `return strJudger.Judge(str)`

File: cluster/router/v3router/judger/list_string_match_judger_test.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package judger
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/config"
```
Comment (gaoxinge) [inline cluster/router/v3router/judger/list_string_match_judger_test.go:21]: move to a new import block, like https://github.com/apache/dubbo-go/blob/master/contributing.md#34-import

File: cluster/router/v3router/judger/method_match_judger_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package judger
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/config"
+	"dubbo.apache.org/dubbo-go/v3/protocol/invocation"
```
Comment (gaoxinge) [inline cluster/router/v3router/judger/method_match_judger_test.go:22]: move to a new import block, like https://github.com/apache/dubbo-go/blob/master/contributing.md#34-import

File: cluster/router/v3router/judger/url_label_match_judge_test.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package judger
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/common"
```
Comment (gaoxinge) [inline cluster/router/v3router/judger/url_label_match_judge_test.go:21]: move to a new import block, like https://github.com/apache/dubbo-go/blob/master/contributing.md#34-import

File: cluster/router/v3router/judger/attachment_match_judger.go (hunk 1 above)
Comment (zouyx) [reply cluster/router/v3router/judger/attachment_match_judger.go:36]: agree

File: cluster/router/v3router/router_chain_test.go [hunk 6]
```diff
@@ -20,6 +20,12 @@ package v3router
 import (
 	"fmt"
 	"testing"
+
+	"dubbo.apache.org/dubbo-go/v3/cluster/router/v3router/k8s_api"
```
Comment (zouyx) [inline cluster/router/v3router/router_chain_test.go:24]: split this block

File: cluster/router/v3router/judger/attachment_match_judger.go (hunk 2 above)
Comment (Mulavar) [reply cluster/router/v3router/judger/attachment_match_judger.go:58]: strJudger.Judge(str) 为 true 的时候需要遍历下一个条件继续做judge，不能直接返回

File: cluster/router/v3router/judger/list_string_match_judger_test.go (hunk 3 above)
Comment (Mulavar) [reply cluster/router/v3router/judger/list_string_match_judger_test.go:21]: this link is 404…

File: cluster/router/v3router/judger/method_match_judger_test.go (hunk 4 above)
Comment (Mulavar) [reply cluster/router/v3router/judger/method_match_judger_test.go:22]: thank u !

File: cluster/router/v3router/judger/attachment_match_judger.go (hunk 2 above)
Comment (gaoxinge) [reply cluster/router/v3router/judger/attachment_match_judger.go:54]: ok

File: cluster/router/v3router/judger/method_match_judger_test.go [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/config"
+	"dubbo.apache.org/dubbo-go/v3/protocol/invocation"
+)
+
+func TestMethodMatchJudger(t *testing.T) {
+
+	methodArgs := make([]*config.DubboMethodArg, 0)
```
Comment (gaoxinge) [inline cluster/router/v3router/judger/method_match_judger_test.go:36]: Remove blank line.

**Review by gaoxinge — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**